test/e2e/fixtures/when.go

```diff
@@ -0,0 +1,74 @@
+package fixtures
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+)
+
+type When struct {
+	t      *testing.T
+	wf     *wfv1.Workflow
+	client v1alpha1.WorkflowInterface
+	name   string
+}
+
+func (w *When) SubmitWorkflow() *When {
+	log.WithField("test", w.t.Name()).Info("Submitting workflow")
+	wf, err := w.client.Create(w.wf)
+	if err != nil {
+		w.t.Fatal(err)
+	} else {
+		w.name = wf.Name
+	}
+	return w
+}
+
+func (w *When) WaitForWorkflow() *When {
+	log.WithField("test", w.t.Name()).WithField("wf", w.name).Info("Waiting")
+	watchIf, err := w.client.Watch(metav1.ListOptions{FieldSelector: fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", w.name)).String()})
+	if err != nil {
+		w.t.Fatal(err)
+	}
+	defer watchIf.Stop()
+	timeout := make(chan bool, 1)
+	go func() {
+		time.Sleep(30 * time.Second)
```

Comment:
Perhaps allow to parametrize this? I can think of some workflows that could take longer that 30 seconds to finish

Reply:
Good idea. I'll add this.